### PR TITLE
feat: Add new credentials for the HTTP Request node

### PR DESCRIPTION
**Pull request by bramkn**
## Summary
Adding Credentials to n8n for easy use in HTTP request node.
Added:
- Datadog
- Filescan
- Elastic -> with API Key.
- Iris Dfir
- Malcore
- Dynatrace -> no test available because of needed scopes for everything.

- https://linear.app/n8n/issue/NODE-1564/review-cred-only-nodes

### CODE REVIEW

**Comment by Joffcom** [comment]
@ShireenMissi if you get a chance could you give this a review, I did an initial review but I have made more changes than the original PR contained I think so it may need a second set of eyes to see if I missed anything.

This is the SecOps nodes that were added as part of the Q2 push.

File: packages/nodes-base/credentials/ElasticSecurityApiKey.credentials.ts
Comment (ShireenMissi) [inline packages/nodes-base/credentials/ElasticSecurityApiKey.credentials.ts:1]: @Joffcom why do we have ElasticSecurityApiKey and ElasticSecurityApi ?

File: packages/nodes-base/credentials/MalcoreApi.credentials.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+} from 'n8n-workflow';
+
+export class MalcoreApi implements ICredentialType {
+	name = 'malcoreApi';
+
+	displayName = 'MalcoreAPI';
+
+	documentationUrl = 'malcore';
+
+	icon = { light: 'file:icons/Malcore.svg', dark: 'file:icons/Malcore.svg' } as const;
```
Comment (ShireenMissi) [inline packages/nodes-base/credentials/MalcoreApi.credentials.ts:15]:
```suggestion
	icon = { light: 'file:icons/Malcore.png', dark: 'file:icons/Malcore.png' } as const;
```

File: packages/nodes-base/credentials/ElasticSecurityApiKey.credentials.ts
Comment (Joffcom) [reply packages/nodes-base/credentials/ElasticSecurityApiKey.credentials.ts:1]: ElasticSecurityApi is for Basic auth and ElasticSecurityApiKey is using an API Key instead.

File: packages/nodes-base/credentials/MalcoreApi.credentials.ts (hunk 1 above)
Comment (Joffcom) [reply packages/nodes-base/credentials/MalcoreApi.credentials.ts]: @ShireenMissi thanks that was a rookie mistake I made 😄

File: packages/nodes-base/nodes/Elastic/ElasticSecurity/GenericFunctions.ts [hunk 2]
```diff
@@ -52,7 +44,8 @@ export async function elasticSecurityApiRequest(
 	}
 
 	try {
-		return await this.helpers.request(options);
+		return await this.helpers.requestWithAuthentication.call(this, 'elasticSecurityApi', options);
+		//return await this.helpers.request(options);
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Elastic/ElasticSecurity/GenericFunctions.ts:48]: do we need this comment?
```suggestion
```

**Review by ShireenMissi — DISMISSED** [review]
LGTM 🎉

**Review by ShireenMissi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.57.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.57.0)

**Pull request merged**